### DISCUSSION TITLE
Node.js Telegram API Error: 400 MEDIA_INVALID in messages.SendMultiMedia

### DISCUSSION THREAD

**Discussion by zareshahi** [gram-js/gramjs#594] [Q&A]
I'm encountering an error in my Node.js application while using the telegram library (gramjs) to interact with the Telegram API. The error message I'm getting is:
``` js
RPCError: 400: MEDIA_INVALID (caused by messages.SendMultiMedia)
at RPCMessageToError (/home/***/node_modules/telegram/errors/index.js:28:12)
at MTProtoSender._handleRPCResult (/home/***/node_modules/telegram/network/MTProtoSender.js:534:58)
...
```
Here's the relevant portion of my code:
``` js
const {Api, TelegramClient} = require("telegram");
const {getSession} = require("../sessions/sessionManager");
const config = require("../config/config");
const {CustomFile} = require("telegram/client/uploads");
const {statSync} = require("fs");
const moment = require('moment');


const session = getSession(); // You should put your string session here
const client = new TelegramClient(session, config.apiId, config.apiHash, {
    connectionRetries: 5,
    proxy: {
        ip: config.proxyHost,
        port: config.proxyPort,
        MTProxy: config.proxyMtproto,
        secret: config.proxySecret,
        socksType: 5,
        // timeout: 10,
    },
});

const test = async (req, res) => {
    await client.connect(); // This assumes you have already authenticated with .start()

    const toUpload = new CustomFile("photo.jpg", statSync("./1700.jpg").size, "./1700.jpg");
    const file = await client.uploadFile({
        file: toUpload,
        workers: 1,
    });
    const media = new Api.InputMediaUploadedPhoto({
        file: file,
    })
    const timeSchedule = moment().add(1, 'hour').unix();
    const request = new Api.messages.SendMultiMedia({
        peer: "test",
        multiMedia: [new Api.InputSingleMedia({
            media: media,
            message: 'test',
        })],
        scheduleDate: timeSchedule
    })
    console.log(request)
    const result = await client.invoke(request)

    console.log(result); // prints the result

};

module.exports = {
    test: test
};
```
I'm attempting to send multiple media files with scheduled sending using messages.SendMultiMedia, but I'm receiving a MEDIA_INVALID error with an HTTP status code of 400. I've already checked the media files, their formats, and my authentication, but I'm still encountering this issue.

Also i saw [this link](https://gram.js.org/tl/messages/SendMultiMedia#messagessendmultimedia) for implementation my code

Could someone please help me understand what might be causing this error and how I can resolve it? Any guidance or insights would be greatly appreciated. Thank you!

**Comment by clackx** [comment]
Same here.
 The [sendMedia](https://gram.js.org/tl/messages/SendMedia) method works, but if I pass the same media to SendMultiMedia method I get a 400 MEDIA_INVALID error.

it works:

```
const myMedia = new Api.InputMediaUploadedPhoto({
    file: await client.uploadFile({
        file: new CustomFile("1.jpg", fs.statSync("./1.jpg").size, "./1.jpg"),
        workers: 1,
    }),
})

const result = await client.invoke(new Api.messages.SendMedia({
    peer: "test",
    media: myMedia,
    message: "Hello there!"
}))
```

it doesn't:
```
const result2 = await client.invoke(
    new Api.messages.SendMultiMedia({
        peer: "test",
        multiMedia: [
            new Api.InputSingleMedia({
                media: myMedia,
                message: "Hello there!",
            }),
        ],
    })
)
```

**Reply by vitaliy21730** [reply]
same problem Оо
Also tested with same files using both sendMedia and sendMultiMedia

**Reply by ifantom** [reply]
Same problem

**Reply by mr-Deviant** [reply]
+1, SendMultiMedia example from docs page doesn't work

**Comment by dmitrypisanko** [comment]
It was tricky, but the answer is  hidden here https://github.com/gram-js/gramjs/blob/5581af24fe041107cac5cf6f88402fa0e05f6b9d/gramjs/client/uploads.ts#L616

First you need to upload the file, call method getInputMedia from Utils to uploaded media and use the result in SendMultiMedia.

**Comment by dathoangit** [comment]
```
await _sendAlbum(client, targetChannel, {
      file: [file1, file2, ...],
      caption: combinedCaption,
      forceDocument: false,
      workers: 1,
    });
```
Upload each media as a CustomFile, collect them into an array, then use _sendAlbum to send them as a single album with a shared caption.

**Comment by guang8886667** [comment]
帮助申请telegram application 秒成功 无论什么账号 解决telegram application error问题

Help apply for telegram application in seconds, no matter what account, solve telegram application error problem

请联系Telegram ： @guang8886667

请联系Telegram : https://t.me/guang8886667
